src/subscriber/invoice-status-subscriber.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ *  SudoSOS back-end API service.
+ *  Copyright (C) 2024  Study association GEWIS
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published
+ *  by the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { EntityManager, EntitySubscriberInterface, EventSubscriber, InsertEvent, UpdateEvent } from 'typeorm';
+import InvoiceStatus from '../entity/invoices/invoice-status';
+import Invoice from '../entity/invoices/invoice';
+
+@EventSubscriber()
+export default class InvoiceStatusSubscriber implements EntitySubscriberInterface {
+  listenTo(): Function | string {
+    return InvoiceStatus;
+  }
+
+  async afterInsert(event: InsertEvent<InvoiceStatus>): Promise<void> {
+    await InvoiceStatusSubscriber.updateInvoiceStatus(event.manager, event.entity.invoice.id, event.entity);
+  }
+
+  async afterUpdate(event: UpdateEvent<InvoiceStatus>): Promise<void> {
+    await InvoiceStatusSubscriber.updateInvoiceStatus(event.manager, event.databaseEntity.invoice.id, event.databaseEntity);
+  }
+
+  static async updateInvoiceStatus(manager: EntityManager, invoiceId: number, status: InvoiceStatus): Promise<void> {
+    const invoiceRepository = manager.getRepository(Invoice);
+    const invoice = await invoiceRepository.findOne({ where: { id: invoiceId }, relations: ['latestStatus'] });
+
+    if (invoice) {
+      invoice.latestStatus = status;
+      await invoiceRepository.save(invoice);
+    }
+  }
```

Comment:
Does this function always return the latest status? What if you update a status? Because then that one will become the latest? What if this status is deleted? Is that case also covered?

Reply:
There are test cases for updating a status (multiple times) and those pass.
You cannot update nor delete a status, we don't have options for those.

But if you would be to update a status (why?) then that would become the latest status. This _might_ not be the cleanest solution, but it is way better than not being able to effectively filter on them.

Reply:
Follow up question: would it then not make sense to include such an extra check? Might indeed take an extra query and a loop to see whether it is indeed the most current status, but it would make the implementation more robust. Let's say we want to implement status "comments" in the future (like PP has), then those changes might cause havoc when you make a typo and fix that.